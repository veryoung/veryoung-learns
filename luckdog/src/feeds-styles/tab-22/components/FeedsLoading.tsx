import React from 'react';
import { View, Image, Animation, StyleSheet } from '@tencent/hippy-react-qb';
import FeedsProtect from '../../../mixins/FeedsProtect';

interface CompProps {
  imgUri: string;
  colors?: string[];
}

@FeedsProtect.protect
export default class FeedsLoading extends React.Component<CompProps> {
  public constructor(props) {
    super(props);
    (this as any).rotateAnim = new Animation({
      mode: 'timing',
      startValue: 0.0,
      toValue: 360.0,
      duration: 1000,
      valueType: 'deg',
      timingFunction: 'linear',
      repeatCount: 'loop',
    });
  }
  public componentDidMount() {
    (this as any).rotateAnim.start();
  }

  public componentWillUnmount() {
    (this as any).rotateAnim.destroy();
  }

  public render() {
    const { imgUri, colors } = this.props;
    return (<View style={styles.loading}>
      <Image style={[
        {
          width: 22.5,
          height: 22.5,
          backgroundColor: 'transparent',
          transform: [{ rotate: (this as any).rotateAnim }],
        },
        colors ? { tintColors: colors } : {},
      ]} source={{ uri: imgUri }} noPicMode={{ enable: false }} reportData={{ sourceFrom: 'FeedsLoading' }} />
    </View>);
  }
}

const styles = StyleSheet.create({
  loading: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
  },
});
