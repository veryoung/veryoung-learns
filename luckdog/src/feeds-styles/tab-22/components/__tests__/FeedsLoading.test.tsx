import FeedsIcon from '../../../../framework/FeedsIcon';
import React from 'react';
import renderer from 'react-test-renderer';

import FeedsLoading from '../FeedsLoading';

describe('<FeedsLoading/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const tree = renderer.create(<FeedsLoading imgUri={FeedsIcon.VideoLoadingCircle} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
