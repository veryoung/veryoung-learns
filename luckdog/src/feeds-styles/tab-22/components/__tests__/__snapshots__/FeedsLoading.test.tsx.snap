// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsLoading/> render testing render normal 1`] = `
<div
  nativeName="View"
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <img
    alt=""
    nativeName="Image"
    noPicMode={
      Object {
        "enable": false,
      }
    }
    reportData={
      Object {
        "sourceFrom": "FeedsLoading",
      }
    }
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAMAAABHPGVmAAAAP1BMVEUAAAD///////////////////////////////////////////////////////////////////////////////9Du/pqAAAAFXRSTlMAzA4askovaXKffr+WWLmoJWCPiEAQ8x8bAAACf0lEQVRo3r2a3ZKDMAiFEfKv0ah9/2fdKXXKTWc2bvfku+qFSuEQogHqhYOPR97cPE3z6vKZamD6T9gXN33AnV7oX5BoBj7gYvvah5qnX3GVv3EiTcZ2Rh+aMBGzBL+XbTKK/NVEeT8jx0AfCPvynRl+e7F47opnYbqJX69470K/0NJ17erpDrJcJjpv8+5ymvp5zGaik/q6pf+OS434Bw0fvVfnl+dCN5FzTb2XvsJbCUhbVY1AQILqdzABaWqjEBLRWCVCwqr5TlAy3g9KeD3ooXlFUOSZWI4JitbdQFC81RIYvH7cDQCZJQRFbP/AUZ6ZRVjEdk4caYAjPMKRqo6AybACz+dWSRHEGjGxm/6KuMVe3svPIWS3iri9c4sJAqsSl7VMGDSnqsUNw37t6A65WQVdgq+wEQQTPGgCwNiecVLdT4JRVPmI0t2UT3RAK/AVqIxKLkuvrMo0giGaww5Vgi2HV5pRlcuMzKRrEYc+f4iRIeECCm/ZhU7hpqVxxGIcUVYOdIHUx6NL/amBGrJpMXY16vY74kUC90pkuhf4y921QAT9msoDXrgd+tNhUb1xH0H2ZPDn3K45hf0wVbUr+BPba27BDgvMkYQ79jBHBHaAY44U8FFUVEdgh2r2vxPseNAW+8rYg86quYQ9sg3aTIIdPltrYRbsMfqhTTRsQ6BoZsFaG/ZvM7ZJs0+qLqrdZH6sgmqcmR5zQ7YA+VAbAdnMDE5j1ZBt2aq3OkE1mK2JnxnXKqc4KQnV9LfZgvkxYHxBUIMYsjsbxMCMlPhlukiMGY6JuXc45v6Yj7Tg47lNRhL0wFKuDB+9EvAQWfG9Ptwfh3NbPqIP3QZ+AFA+E3UIPjJ9AAAAAElFTkSuQmCC"
    style={
      Object {
        "0": Object {
          "backgroundColor": "transparent",
          "height": 22.5,
          "transform": Array [
            Object {
              "rotate": Animation {
                "animationId": undefined,
                "delay": 0,
                "destroy": [Function],
                "direction": "center",
                "duration": 1000,
                "inputRange": Array [],
                "mode": "timing",
                "onHippyAnimationCancel": [Function],
                "onHippyAnimationEnd": [Function],
                "onHippyAnimationRepeat": [Function],
                "onHippyAnimationStart": [Function],
                "onRNfqbAnimationCancel": [Function],
                "onRNfqbAnimationEnd": [Function],
                "onRNfqbAnimationRepeat": [Function],
                "onRNfqbAnimationStart": [Function],
                "outputRange": Array [],
                "repeatCount": -1,
                "startValue": 0,
                "timingFunction": "linear",
                "toValue": 360,
                "valueType": "deg",
              },
            },
          ],
          "width": 22.5,
        },
        "1": Object {},
      }
    }
  />
</div>
`;
